Accessibilty_Scanner_IAM (#9899)
braze-inc/braze-docs: _docs/_user_guide/message_building_by_channel/in-app_messages/testing.md

- Is your copy clear, concise, and correct?
- Do your buttons direct the user where they should go?

## Accessibility Scanner

![Accessibility scan results]({% image_buster /assets/img/Accessibilty_Scanner_IAM.png %})

### Overview

> To support accessibility best practices, Braze offers an Accessibility Scanner that automatically reviews the content of in-app messages created using the traditional HTML editor. This tool is designed to assist with identifying common accessibility issues, making it easier to work toward compliance with modern standards.

Reviewer: Are we treating this like a branded feature? 🤔 I would assume not, so would suggest not capitalizing it. We can also speak more directly to the action here imo, since there is nothing the user needs to do for this to occur. I would also suggest removing the custom HTML bit from the bottom and making that more explicit/clear at the top here instead.

```suggestion
To support accessibility best practices, Braze automatically scans the content of in-app messages created using the traditional HTML editor against accessibility standards. This scanner helps identify content that may not meet [WCAG 2.1 AA standards](https://www.w3.org/WAI/standards-guidelines/wcag/), the internationally recognized guidelines for making web content more inclusive for people with disabilities.

This check only applies to custom HTML IAMs—it does not currently support messages built with the drag-and-drop editor.
```

> To support accessibility best practices, Braze offers an Accessibility Scanner that automatically reviews the content of an IAM messages, when created using the traditional HTML editor . This tool is designed to assist with identifying common accessibility issues, making it easier to work toward compliance with modern standards.

Reviewer 2: ```suggestion
To support accessibility best practices, Braze offers an Accessibility Scanner that automatically reviews the content of in-app messages created using the traditional HTML editor. This tool is designed to assist with identifying common accessibility issues, making it easier to work toward compliance with modern standards.
```

> ### How It Works

Reviewer: ```suggestion
### How it works
```

> When a Custom HTML message is created, the Accessibility Scanner analyzes the message body and checks it against the entire set of WCAG 2.1 Level AA guidelines. It highlights known issues and provides:
> 
> - The HTML element(s) violated
> - A description of the rule violation
> - Link to learn more about the violation, potentially includuing guidance on how to address the issue

Reviewer: ```suggestion
The scanner runs automatically and evaluates your full HTML message against the full [Web Content Accessibility Guidelines](https://www.w3.org/WAI/standards-guidelines/wcag/) (WCAG) 2.1 AA rule set. For each flagged issue, it shows:

- The specific HTML element involved
- A description of the accessibility issue
- A link to additional context or remediation guidance
```

### Important Notes

> - Not Comprehensive: The Accessibility Scanner is intended as a helpful diagnostic tool, not a complete accessibility audit. It may not catch every possible issue.

Reviewer 2: this really is jut not true

Reviewer: I agree that we can definitely fine tune the language here, as the checks are exhaustive. I think something to hedge against here (and this might be what the sentence is currently going for) is just that automated accessibility testing is not a complete picture of message accessibility, as not all tests can be done automatically. 

To accommodate for this in the Inbox Vision docs, we added a section on [Understanding automated accessibility testing](https://www.braze.com/docs/user_guide/message_building_by_channel/email/inbox_vision/#understanding-automated-accessibility-testing). Perhaps what we can do is slightly generalize this section and make it into an includes snippet so we can reuse it across articles. (If we do this, @lydia-xie perhaps we move the callout on document `<title>` up to the "Serious" subsection, since that is what that rule is categorized as?) That would cover all of the important notes here except for custom HTML, which imo should be at the very top.

If this sounds good I can work w/ Lydia to make that update to this PR.

Author: @bre-fitzgerald I like the idea of linking this to the main accessibility article.  Could we do both 1 and 2? 

1) Generalize this section and include snippets
2) Link, where there might be redundant information to the main accessibility article?

~Gurbir

Reviewer: Yup that makes sense to me. I'll add this to my to do list for tomorrow! 
cc: @lydia-xie

Reviewer: Just pushed some updates here with these changes! Please review and let me know if you have any additional feedback. I also adjusted a few link references and added inks out to the [filtered list of WCAG rules](https://www.w3.org/WAI/WCAG22/quickref/?versions=2.1&currentsidebar=%2523col_customize&levels=aaa) for each ruleset (2.1 and 2.2) in the respective articles.

- Is your copy clear, concise, and correct?
- Do your buttons direct the user where they should go?

## Accessibility scanner

To support accessibility best practices, Braze automatically scans the content of in-app messages created using the traditional HTML editor against accessibility standards. This scanner helps identify content that may not meet Web Content Accessibility Guidelines ([WCAG](https://www.w3.org/WAI/standards-guidelines/wcag/)) standards. WCAG is a set of internationally recognized technical standards developed by the World Wide Web Consortium (W3C) to make web content more accessible to people with disabilities.

![Accessibility scan results]({% image_buster /assets/img/Accessibilty_Scanner_IAM.png %})

{% alert note %}

> The in-app message accessibility scanner only runs on messages built with custom HTML. 

Reviewer: ```suggestion
The in-app message accessibility scanner only runs on messages built with the HTML editor.
```
